### PR TITLE
Truly disable thumb emission

### PR DESCRIPTION
**Pull request by aschwaighofer**
Interpreter support for thumb is not yet properly implemented/tested.
Disable thumb emission.

This change is necessary because LLVM now uses the features string in the
metadata. This features string contains '+thumb'.

This is a follow-up to 034241e440ec6b4782eba22d3f12a41e47b0b8cb.

rdar://32599805
rdar://34781037 (tests that expect non-thumb instructions are failing)

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

File: lib/IRGen/IRGenModule.h [hunk 1]
```diff
@@ -1120,6 +1120,9 @@ class CurrentIGMPtr {
   IRGenModule *operator->() const { return IGM; }
 };
 
+/// Workaround to disable thumb-mode until debugger support is there.
+bool shouldRemoveTargetFeature(const std::string &);
```
Comment (jrose-apple) [inline lib/IRGen/IRGenModule.h]: Nitpick: StringRef

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

**Comment by gparker42** [comment]
Are we unconditionally disabling Thumb, or only disabling Thumb for debugger JIT use? Removing Thumb from static compilation will be terrible for code size.

**Comment by aschwaighofer** [comment]
Yes this is bad for code size. This is unconditionally. We never emitted thumb for Swift code.

**Comment by jrose-apple** [comment]
rdar://problem/32599805

**Comment by gparker42** [comment]
Okay, I thought we had shipped something with Thumb enabled.

**Pull request merged**